src/AppInstallerRepositoryCore/Public/winget/RepositorySearch.h

```diff
@@ -178,11 +178,24 @@ namespace AppInstaller::Repository
         TrackingWriteTime,
         // The Architecture of an installed package
         InstalledArchitecture,
+        // The PackagePinnedState of the installed package
+        PinnedState,
     };
 
     // Convert a PackageVersionMetadata to a string.
     std::string_view ToString(PackageVersionMetadata pvm);
 
+    // Possible pinned states for a package.
+    // Pinned packages need to be explicitly updated (i.e., are not included in `upgrade --all`)
+    enum class PackagePinnedState
+    {
+        Unknown,

```

Comment:
I would suggest that unknown and "not pinned" are the same state, then use `NotPinned` as the name to better indicate that.